src/App.js

```diff
@@ -1187,9 +1192,13 @@ class App extends Component {
     // @ts-ignore this should be ok when currentPhraseAttempts is typed correctly instead of never[]
     const currentPhraseAttempts = this.state.currentPhraseAttempts.map(copy => ({...copy}));
 
+    if (buffer) {
+      // Assuming a single buffer contains inputs from one stroke, peaks are always at the end of buffer. (i.e. steno can either keep typing or backspace first and then keep typing) Since currentPhraseAttempts is only used to calculate attempts and its logic is to use peaks or the last strokes, it's safe to push elements that lacks `numberOfMatchedWordsSoFar` and `hintWasShown` into `currentPhraseAttempts`.
```

Comment:
This might be questionable, so providing some supporting fact: world fastest clicker of mouse is 14 times/s, that's 71ms. You'd need 4+ timers faster strokes than that to put two strokes within 16ms window

https://www.google.com/search?q=how+quickly+can+human+click&sca_esv=78be3f8d90fd2ae1&sca_upv=1&sxsrf=ACQVn095ib1YM6jvNx-99TtApmyqXBqIKg:1714270579476&ei=c7EtZqHTHJbj2roP5d2MmAk&ved=0ahUKEwjh2MDH6-OFAxWWsVYBHeUuA5MQ4dUDCBA&uact=5&oq=how+quickly+can+human+click&gs_lp=Egxnd3Mtd2l6LXNlcnAiG2hvdyBxdWlja2x5IGNhbiBodW1hbiBjbGljazIIECEYoAEYwwRIwQ9QvwJY2w1wAngBkAEAmAGYAaAB6QaqAQMwLje4AQPIAQD4AQGYAgagAv4DwgIKEAAYsAMY1gQYR8ICBhAAGAgYHsICCxAAGIAEGIYDGIoFwgIIEAAYgAQYogTCAgYQABgHGB7CAgcQABiABBgNmAMAiAYBkAYIkgcDMi40oAf7FQ&sclient=gws-wiz-serp

Reply:
One random weird quirk is that there can sometimes be a single stroke with a more complex Plover translation that changes direction of growing or shrinking the output text length more than once, such as `=retro_toggle_asterisk`. If a student has an entry defined like `"T*EFT": "=retro_toggle_asterisk"` then writes `H-D/T*EFT` in 2 strokes, it would output `had had` then `Hodgkin's disease`. They could also write `H-D/*/H*D` in 3 strokes and the output would look similar.

I don't think this changes much for us but I wanted to note it.

Reply:
Thanks for the heads up! I didn't know such a command which can totally replace existing text. I agree that with that command, there will be still one peak in a single stroke, unless Plover only deletes at the beginning of the input of a single stroke (i.e. input chars are deleted immediately, which is unnecessary)